### ISSUE TITLE
New loss function: ROC AUC score

### ISSUE THREAD

**Issue by saddy001** [lmjohns3/theanets#129]
It would be very nice to have the [ROC AUC](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html) as classifier loss. Theanets has hidden the tensor logic from me so successfully, that I'm not able to write the loss function on my own.

![rmsprop](https://cloud.githubusercontent.com/assets/13658554/15157455/059ec010-16ec-11e6-9a8f-ebcbd556aa96.png)

**Comment by saddy001** [comment]
Someone has already [done the work](https://github.com/lisa-lab/pylearn2/blob/master/pylearn2/train_extensions/roc_auc.py) for pylearn2.
